specification/compiler/model/utils.ts

```diff
@@ -422,13 +384,137 @@ export function modelProperty (declaration: PropertySignature | PropertyDeclarat
 
   // names that contains `.` or `-` will be wrapped inside single quotes
   const name = declaration.getName().replace(/'/g, '')
-  return {
+  const property = {
     name,
     required: !declaration.hasQuestionToken(),
     type: modelType(type)
   }
+  hoistPropertyAnnotations(property, declaration.getJsDocs())
+  return property
+}
+
+/**
+ * Pulls @obsolete and @obsolete_description from types and properties
+ */
+function setObsolete(type: model.Depreciable, tags: Record<string, string>) {
+  const obsolete = tags['obsolete']
+  const description = tags['obsolete_description']
+  if (obsolete) {
+    type.deprecation = { version: obsolete, description: description}
+  }
+  delete tags['obsolete']
+  delete tags['obsolete_description']
 }
 
+/**
+ * Validates ands sets jsDocs tags used throughout the input specification
+ */
+function setTags<TType extends model.BaseType | model.Property | model.EnumMember>(
+  type: TType,
+  jsDocs: JSDoc[],
+  validTags: string[],
+  setter: ((tags: Record<string, string>, tag:string, value:string) => void)
+) {
+  if (jsDocs.length == 0) return
+  const tags = parseJsDocTags(jsDocs)
+  if (Object.keys(tags).length == 0) return
+
+  if (model.isDepreciable(type)) {
+    setObsolete(type, tags)
+  }
+  const badTags = Object.keys(tags).filter(tag => !validTags.includes(tag))
+  assert(
+    badTags.length == 0,
+    `'${getName(type)}' has the following unknown annotations: ${badTags.join(', ')}`
+  )
+
+  for (const tag of validTags) {
+    if (tag == 'behavior' && model.isBehaviorAttachable(type)) continue
+    if (tags[tag]) {
+      setter(tags, tag, tags[tag])
+    }
+  }
+
+  function getName(type) {
+    if (type instanceof model.BaseType) return type.name
+    if (type instanceof model.Property) return type.name
+    if (type instanceof model.EnumMember) return type.name
+    return typeof(type)
```

Comment:
Probably I'm missing something, what is the `typeof` function?

Reply:
leftover :) it always returns `[Object]`